CMakeLists.txt

```diff
@@ -46,30 +46,30 @@ set (COMMON_WARNING_FLAGS "-Wall -Werror")
 set (CXX_WARNING_FLAGS "-Wnon-virtual-dtor -Wold-style-cast")
 
 set (CXX11_ABI "DEFAULT" CACHE STRING "Use C++11 ABI: DEFAULT, ENABLE, DISABLE")
-set (TEST_COVERAGE FALSE CACHE BOOL "Enables flags for test coverage")
-set (ENABLE_TESTS TRUE CACHE BOOL "Enables tests")
+option (TEST_COVERAGE "Enables flags for test coverage" OFF)
```

Comment:
Let merge this PR containing mixture of different changes and refine separate changes lately.